debezium-server/debezium-server-http/src/main/java/io/debezium/server/http/HttpChangeConsumer.java

```diff
@@ -0,0 +1,121 @@
+@Named("http")
+@Dependent
+public class HttpChangeConsumer extends BaseChangeConsumer implements DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpChangeConsumer.class);
+
+    private static final String PROP_PREFIX = "debezium.sink.http.";
+    private static final String PROP_WEBHOOK_URL = "url";
+    private static final String PROP_CLIENT_TIMEOUT = "timeout.ms";
+
+    private static final Long HTTP_TIMEOUT = Integer.toUnsignedLong(60000); // Default to 60s
+
+    private static String sinkUrl;
+    private static Duration timeoutDuration;
+    private static String contentType;
+
+    private HttpClient client;
+    private HttpRequest.Builder requestBuilder;
+
+    // If this is running as a Knative object, then expect the sink URL to be located in `K_SINK`
+    // as per https://knative.dev/development/eventing/custom-event-source/sinkbinding/
+    @PostConstruct
+    void connect() throws URISyntaxException {
+        client = HttpClient.newHttpClient();
+        final Config config = ConfigProvider.getConfig();
+        String sink = System.getenv("K_SINK");
+        timeoutDuration = Duration.ofMillis(HTTP_TIMEOUT);
+
+        if (sink != null) {
+            sinkUrl = sink;
+        }
+        else {
+            sinkUrl = config.getValue(PROP_PREFIX + PROP_WEBHOOK_URL, String.class);
+        }
+
+        config.getOptionalValue(PROP_PREFIX + PROP_CLIENT_TIMEOUT, String.class)
+                .ifPresent(t -> timeoutDuration = Duration.ofMillis(Long.parseLong(t)));
+
+        switch (config.getValue("debezium.format.value", String.class)) {
+            case "avro":
+                contentType = "avro/bytes";
+                break;
+            case "cloudevents":
+                contentType = "application/cloudevents+json";
+                break;
+            default:
+                // Note: will default to JSON if it cannot be determined, but should not reach this point
+                contentType = "application/json";
+        }
+
+        LOGGER.info("Using http content-type type {}", contentType);
+        LOGGER.info("Using sink URL: {}", sinkUrl);
+        requestBuilder = HttpRequest.newBuilder(new URI(sinkUrl)).timeout(timeoutDuration);
+        requestBuilder.setHeader("content-type", contentType);
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, DebeziumEngine.RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        for (ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+            String value = "";
+            if (record.value() != null) {
+                value = (String) record.value();
+
+                HttpRequest request = requestBuilder.POST(HttpRequest.BodyPublishers.ofString(value)).build();
+                client.sendAsync(request, HttpResponse.BodyHandlers.ofString())
```

Comment:
`sendAsync` might change the ordering of events, is it ok?

Reply:
That's a good point; no, in general we should not send events out of order, as for instance an INSERT and a DELETE event for the same record received in the wrong order would leave a consumer in a wrong state. We could mimick Kafka's semantics and e.g. work with multiple clients for different topics/partitions; but perhaps for now we just stick to sync sending of everything?

Reply:
> That's a good point; no, in general we should not send events out of order, as for instance an INSERT and a DELETE event for the same record received in the wrong order would leave a consumer in a wrong state. We could mimick Kafka's semantics and e.g. work with multiple clients for different topics/partitions; but perhaps for now we just stick to sync sending of everything?

I think using the sync client is a good start.

In addition, I'm expecting that CloudEvents emitted by Debezium contain the [CloudEvent's partitioning extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/partitioning.md) to preserve ordering across multiple hops, otherwise, the ordering wouldn't be guaranteed in all cases, so the question is, are we already setting this extension?

FYI, we have created the logic to do ordered delivery over HTTP on top of a single async Vert.x's WebClient in Knative, maybe moving forward we can extract that logic into a reusable package that can be consumed by both Debezium and Knative (if using the Vert.x's Webclient in Debezium is an option)

Reply:
Funny, I did that with the original Apache HTTP Client approach, and opted to swap it out for async figuring `iodebeziumlsn`. Upon double checking the test data, this value is the same, and can see the ordering aspect being an issue, and can do this as a fix.

One question I do have, while I originally used the Apache HTTP Client, I was encouraged to swap that out for the jdk-11 built in HTTP client. Wouldn't the project be going with that going forward?

Reply:
As to the CloudEvent partitioning extension, this is the one attribute I did not observe in my testing. However, wouldn't the partition key need to be known prior to sending the events? Granted easier to configure when using Kinesis, Kafka, or EventHub.

Reply:
I think we should stick to the Java 11 built-in client; I'd assume it has means of sending sync requests?

Reply:
It does and have updated the PR to reflect that.

Reply:
> As to the CloudEvent partitioning extension, this is the one attribute I did not observe in my testing. However, wouldn't the partition key need to be known prior to sending the events? 

Yes, I'm assuming Debezium knows the partition key (?)